src/Home.js

```diff
@@ -1,26 +1,116 @@
 import React from 'react';
 import Axios from 'axios';
+import ImageMapper from 'react-image-mapper';
 
 class Home extends React.Component {
 
     constructor(props) {
         super(props)
+        this.state = {
+            hoveredArea: '',
+            routeData: [],
+            URL: "./GymOverview2.png"
+        }
+        this.getActive = this.getActive.bind(this)
+
+
+
+
+    }
+    getActive() {
+        //check localstorage first, if not in localstorage, run axios, else set state from localstorage
+        Axios.get('http://localhost:8000/api/show')
+            .then(res => {
+                const data = res.data
+                localStorage.setItem('routes', JSON.stringify(data))
+                this.setState({ routeData: data })
+            })
+    }
+
+    componentDidMount() {
+        this.getActive();
+    }
```

Comment:
I think components are generally easier to read and understand if lifecycle functions like these are at the top, followed by your own component functions.